src/engine/Config.java

```diff
@@ -0,0 +1,96 @@
+package engine;
+
+import data.DataType;
+import errors.NoDataException;
+import errors.NoSortException;
+import utils.SortType;
+import utils.Utils;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+
+public class Config {
+
+    private final static String DATA_TYPE = "-dataType";
+    private final static String SORT_TYPE = "-sortingType";
+    private final static String WRITE_DATA = "-outputFile";
+    private final static String READ_DATA = "-inputFile";
+
+    private DataType type;
+    private SortType sortType;
+    private File inputFile;
+    private File outputFile;
+
+    public Config(String[] args) {
+        init(Utils.parseArgs(args));
+    }
+
+    public SortType getSortType() {
+        if (sortType == null) {
+            throw new NoSortException();
+        } else {
+            return sortType;
+        }
+    }
+
+    public DataType getType() {
+        if (type == null) {
```

Comment:
If command is incorrect it is better to fail as soon as possible. That is why his check and throwing of exceptions should be on the validation step.